index.bs

```diff
@@ -1370,77 +896,768 @@ What configuration is the device in after it resets?
+        the first |bytesTransferred| bytes of |buffer| and set
+        <code>|result|.{{USBInTransferResult/data}}</code> to a new {{DataView}}
+        constructed over it.
+    1. If <a>this</a> responded with more than |length| bytes of data set
+        <code>|result|.{{USBInTransferResult/status}}</code> to {{"babble"}}.
+    1. If the transfer ended because |endpoint| is halted set
+        <code>|result|.{{USBInTransferResult/status}}</code> to {{"stall"}}.
+    1. If <a>this</a> acknowledged the complete transfer set
+        <code>|result|.{{USBInTransferResult/status}}</code> to {{"ok"}}.
+    1. If the transfer failed for any other reason <a>reject</a> |promise| with a
+        {{NetworkError}} and abort these steps.
+    1. <a>Resolve</a> |promise| with |result|.
+</div>
+
+<div algorithm="USBDevice.transferOut(endpointNumber, data)">
+    The <dfn for=USBDevice method>transferOut(|endpointNumber|, |data|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the following steps
+    <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
+    1. Let |endpointAddress| be |endpointNumber| (i.e {{USBDirection/"out"}} direction).
+    1. Let |endpoint| be the result of <a>finding the endpoint</a> with
+        |endpointAddress| and <a>this</a>.
+    1. If |endpoint| is <code>null</code>, <a>reject</a> |promise| with a
+        {{NotFoundError}} and abort these steps.
+    1. If |endpoint|.{{USBEndpoint/type}} is not equal to {{USBEndpointType/"bulk"}},
+        <a>reject</a> |promise| with an {{InvalidAccessError}} and abort these steps.
+    1. As appropriate for |endpoint| enqueue a <a lt="bulk transfer">bulk</a> or
+        <a lt="interrupt transfer">interrupt</a> <a>OUT transfer</a> on |endpoint|
+        to transmit |data| to the device.
+    1. Let |result| be a new {{USBOutTransferResult}}.
+    1. Set <code>|result|.{{USBOutTransferResult/bytesWritten}}</code> to the
+        amount of data successfully sent to the device.
+    1. If the endpoint is stalled set
+        <code>|result|.{{USBOutTransferResult/status}}</code> to {{"stall"}}.
+    1. If the device acknowledges the complete transfer set
+        <code>|result|.{{USBOutTransferResult/status}}</code> to {{"ok"}}.
+    1. If the transfer fails for any other reason <a>reject</a> |promise| with a
+        {{NetworkError}} and abort these steps.
+    1. <a>Resolve</a> |promise| with |result|.
+</div>
+
+<div algorithm="USBDevice.isochronousTransferIn(endpointNumber, packetLengths)">
+    The <dfn for=USBDevice method>isochronousTransferIn(|endpointNumber|, |packetLengths|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the following
+    steps <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
```

Comment:
```suggestion
        and abort these steps if |promise| is rejected.
```

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/5aaf696938ed1dc073f8cc8da8c809e9e5019f41.